app/controllers/integrated/introduce_yourself_controller.rb

```diff
@@ -14,20 +12,18 @@ def update_models
                            month: member_data.delete(:birthday_month),
                            year: member_data.delete(:birthday_year),
       ))
-      if current_application
+      if current_application.primary_member.is_a?(HouseholdMember)
         current_application.primary_member.update(member_data)
         current_application.update(application_params)
       else
-        application = CommonApplication.create(application_params)
-        application.members.create(member_data)
-        session[:current_application_id] = application.id
+        current_application.members.create(member_data)
       end
     end
 
     private
 
     def existing_attributes
-      if current_application
+      if current_application.primary_member.is_a?(HouseholdMember)
```

Comment:
I feel iffy about this. Because `current_application.primary_member` will return a `NullMember`, this check (or something like it) is needed.